### PR TITLE
Remove sourceFullPath from Unix FileSystem.MoveDirectory exception

### LINKED ISSUE #50833: Directory.Move throws invalid exception message
### When _destination_ directory is invalid, Directory.Move() throws an exception claiming that parts of _source_ directory does not exist.

Snippet:
```
using System.IO;

try
{
    // cleanup
    Directory.Delete("dst", true);
} catch {}
Directory.CreateDirectory("src");
Directory.Move("src", "dst/dst"); // dst/dst is nested and doesn't exist ; move failure
```

Current error message:
> Unhandled exception. System.IO.DirectoryNotFoundException: Could not find a part of the path '/home/dev/projects/test/src'.
>    at System.IO.FileSystem.MoveDirectory(String sourceFullPath, String destFullPath)
>    at System.IO.Directory.Move(String sourceDirName, String destDirName)
>    at <Program>$.<Main>$(String[] args) in /home/dev/projects/test/Program.cs:line 8

I think the exception message should be something like: `System.IO.DirectoryNotFoundException: Could not find a part of the path '/home/dev/projects/test/dst/dst'.`

I guess the error comes from here: https://github.com/microsoft/referencesource/blob/5697c29004a34d80acdaf5742d7e699022c64ecd/mscorlib/system/io/directory.cs#L1266

### Configuration

.Net 5.0.102 - Ubuntu 20.4 - x64

### PR DESCRIPTION
**Pull request by steveberdy**
## Summary

Removes the `sourceFullPath` argument from the exception thrown when a destination file path for `Directory.Move` has a parent that doesn't exist in Unix. This matches the functionality of Windows. The results should be more consistent and less confusing exceptions from the `Directory.Move` method on both Unix and Windows platforms.

For reference, see the first possible solution for this problem https://github.com/dotnet/runtime/issues/50833#issuecomment-841808221:
> I see 2 possible solutions:
> 
> * Remove the path from the error message of the exception, as done on Windows side.
> * Add extra checks (on Windows and Unix sides) to ensure the parent (destination) directory exists.

### Other minor fixes
 - Typos in documentation fixed

Fixes #50833

### CODE REVIEW

**Comment by stephentoub** [comment]
> Multiple if-statements were condensed to one switch, making code more performant

For reference, not really.  The compiler's still going to generate cascading ifs, similar to the original source code:
https://sharplab.io/#v2:EYLgtghglgdgPgAQEwEYCwAoBAGABAlAOgEkB5Abk0wQGZ8lcBhXAb01w/zoQBZcBZABSwALrgCmAJ0kB7SYxkATcQBpcAZRGTYAc1wBnGQFdJAY3EAxIwBtrABQgiAFms3aYe5fpFXbD5wCUrOycofoA7lAipk64glKy8kriASGhHGwY6dm4phD64rjEMCJSMgAOhACi0nL61QBKDaQNAPoWxAAyVa0AcqQAKu2kAKq9ACIgaTnZzrLhuADqsDRI/BCS+k4Q1oQA4uIiVQAe5uUiUDIwFnLLMKs1iYLFpbKVj3WNzW12AIIDAAk+oNhmNxmpDCZzL57I4nAFKFkZpw8gUiiUyu9apsvi1Wr9Og0qr9xgBNVpVAAaxHUA3UUyRyPScxkCzuq3Wm22uwOR1O4nOl2utxWSA+kmeGLe1Wx9SqTTxBKJJPJVJpdLUXh8NlhgURTI4AHpDbgBk4oPpck5xKYANa4cL5XByKA6WA7awAT1w5SMYlguAAZnIlrAAJzHOJGGDByQiaOOcReh1RJzGMT6co2qA7XL5XS4KK4EQyXDAQqx0MwCPOmBegKEJaFPIwADkYhiEA8hWchV6A1yyXKcNwMFZQZDwAgdtMMjA5VwknE+Su9WmM1RhRemJliTlCrav0YjCq6nUrXGVV6xCqk1wxqr/eLTnmlrIJzOFyujbNFqtNvtP9KzuCNG0WZsu3bK0ux0Qoi0rKcZznSp1wNFkFhgcQFnfflBSuQR1AaQgbkkSARAIoiRhgCA/TTbQAC9xEUX5THMfR9FaMhxnEGAoCY1p/BcAxjDMSwdUEgI1HZNYNi2HZCHWW1xABBoLFkMBxQUZR4mxLSUgRVDsmUQMaOsEQGQNZkX3HaTOTknlDg/AUv2FSRpPFHTEj0gzGXSABfaYAowILqBQAA2egNAaTBMlCWh8HChK8ComjnBdRjmNY8R2M40huN4/jBNwABeAA+XAACIKv1ThRCkajrASiKCDwEiyMEFrcATWj0qYli2I4rieL4xQBLhNROo2HQgjK595lHLDcF6GQRGIedrHEMAeNKRQnLwmBBB8kKsES5Aqw5WTuRi6Y6skBqmtwPaXNwXknqFEj3OxYQSgkbEZvK9CFoWZbVvWzbtqYt78J80JbvuggIqhmAXsc3CXI+0UPNEX7EjUGRgAAKxtMRh0CEqAesjDFpBtbyg2raMV2tGhUOxEbqleHEuxxTlNU9TNOSb6xASORvOmWKckBzDgZW2n6YhpnPxZmHOCC46Eci7c3mu3y4dzDXhzjHNGrO8V9HFwzCw53NZxgbwrbEeVvladQRmPU91BK3BsGObAauyPXGtt+3sadvFiF6AA1AliHGdoxkYAYyF6L2fZQf30kDwc7f9H6w7aDpumBIYLFGCZU+OJAM9h62g9XXPHYPAT/iBfoS7L8YK5oavatr7OQ7zpujxPM8LyvG9O+K73jgAVh7jgs+DhvHqbiPo86WPWgBX4Jm6Cuwvnh36pt+uHZX5224pfoRj2IF+CqfgWlJCuAA5D8X0/Q9XqOY7j8Z/l+BXcY78+5LzPvnTiP8N5/waMQSOVQK4WBAa8e6YCv4X1IK0R+RJ2hdFPBXFAVdLYfxzuApul9lRkgIXPYhoDP6D2dgAIRJK0RgpB+D8B3pPae4VkHHzrqQ9BeJmFx26L0PYgICFv1oSgk+giGF4nUNvWB4jWiRzIJ0f4ycK5ID9jI/h/dl4QM6KQRgABpNRGitGkBTlPH2qA+F3TkQPRuztt67x6FUUgFgdEH30U4gRLjz54kvq7OwdgWgDFvF3Ihvka6yMCUYpuIi+hVAGH8SRdjjg0BoXE3uCTDFkIvmkxYLQLHDw9mPa80Ssk8HTv41B9DXHCJYb0NJfRfj3wrjwbuDTnFJOdoXTx1JaSeyyTPPReSF50Pkc0toa9f7NwaJ0tJ8oK4zwAOyOMabM4JbRGHNDMVeASxA7AIKyWFYBfTElFLxOMGkFiLAjE6J0CuGzJkzBIUEiB6gH6tCTvfUYA4skbLDNs/pty2iMAJJ0YunF+B2G6PffsNTp4bOkVMo+ATClCPmb0V2FgOiMBvP2VojCRiErWSC344Kbm4sgevTevQVlvMYbSnFCi2htL2Fo+BLsqhVDMa/XpmKvkDLucQLlIIH52AGM/LJYZ6mipmd85JLCMkAmZV0rJvwlWfJVeKtoJjzHtF+Hg7hPtfhbOuRyuZZKRjqHldPX4NKbVoM5fiQkxIyQUhGXSCujDrXKoKe6u1KSqQ9FLg0ThwLp6MD1TkMVkKr6R2jlK9uYIK6MDZW6ppezcFiJWb6k8cdlniPOXGqo7LQ35ojVgo8AII49DlWcrBNIY2MABEAjF+qQ15ogUM/5pBMGaIaHsCtPtxhINzbsiBdhTk9HJY6iuVQg29oMTWiB/QLwAJXT2xNBrk3ztbZfNhvQ2mJ1RT7KoYKZ2qudtgno/8BiAKycSat/byGYMfZxKJ/Axk8OwCK9d2LN1N3uUSROT8CHYBzcGjdn6MFDFIIsNpDQCG6I/bOpuAxh31t6OSCJoyMOuvg6BxDeI/gNCTgSVhpA7BOrTkgODIGdn3sVLAwE98k6MFaKQeBqkTGLB0bwu9hqTmtrPReqJFrK4oDXQevt2HnbHp6BvWk49xEif3QHQ99L+AjBfaSje/BiBDCpCeW8V7K48BY4phDym8T0flNY3o+JGGROszQKowH7Pkcc/MzBEc2EIu6FEruVQeBYfY4FgSV57laayd53JrGIX0u3bho5tjp40EDB8vzbHxO/IaOok8F5SCnjhWqWk3SkBRbE8m7dYxiBsMvP8st6hNFJxsd0met6yOFeTeMF5kCzOmvNb10jqW6UerYWMKJbQAVeMM90sKCbdNKZi8CfEZ5THEFc90l+vmNsOa2+MNrFgzWvNqS/erA20seuG7Cy+pdM03ZSwVh7dqXsd26WGfLJ3/NbbaQMUpDQLFjGVJ234jC961MYNO+7M3vs2PxMeMuQx1Cx26Te6L4nL4wp21jvYbTZMzxQHd6btr81jDMf0VDrQiTqKxz18ZKB+tU7A87BZ0C+OoapdPcndnAeDfpTzzedhYGcIaOSPYzQRh2HWSgRgePGuYNdp2gSsD1HdHHeshxDX6WS7gXg8dcKARVE6GTpAsTOcUbxYyuOR5ruC5oDpzOemPXi7jr8xgIxOPkkvOoRg6Hxk0Cm595H+bvcu2x2HjnkfqcQJScQBF8p1A2Nc60bo8CXc+xnjwW3ieud4mhaS5zbmd42NJI/B16yemq/S+rqofuA98bc6QRhAApFvsb888D8UjpPTcy+X1T2choGfmXhfGTPd38TTvicYJo0eSpvWqj9QB/PL8i8i6+/m7djqNP8EZ6eUYDQR7rJdY3j1S2gXrMYZT4v9uKTwP7CYvYBaejQ/LWTqdN+7UBgGgHUZNT8utk4lFTlxtOh/cJ1jgwoLArkh8S82giQM9/cytNFVEfs3s0V5MACD8QRfhT8qNfkBJSAI4+9jgUBCFB9Qg1YgA=

File: src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Windows.cs [hunk 1]
```diff
@@ -123,17 +123,18 @@ public static void MoveDirectory(string sourceFullPath, string destFullPath)
...
+                    case Interop.Errors.ERROR_FILE_NOT_FOUND:
+                        throw Win32Marshal.GetExceptionForWin32Error(Interop.Errors.ERROR_PATH_NOT_FOUND, sourceFullPath);
+                    case Interop.Errors.ERROR_ALREADY_EXISTS:
+                        throw Win32Marshal.GetExceptionForWin32Error(Interop.Errors.ERROR_ALREADY_EXISTS, destFullPath);
+                    // This check was originally put in for Win9x (unfortunately without special casing it to be for Win9x only). We can't change the NT codepath now for backcomp reasons.
+                    case Interop.Errors.ERROR_ACCESS_DENIED: // WinNT throws IOException. This check is for Win9x. We can't change it for backcomp.
+                        throw new IOException(SR.Format(SR.UnauthorizedAccess_IODenied_Path, sourceFullPath), Win32Marshal.MakeHRFromErrorCode(errorCode));
+                    default:
+                        throw Win32Marshal.GetExceptionForWin32Error(errorCode);
+                }
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Windows.cs:137]: Let's revert this file.  It's just a stylistic change.

File: src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Unix.cs [hunk 2]
```diff
@@ -405,7 +405,7 @@ public static void MoveDirectory(string sourceFullPath, string destFullPath)
                     case Interop.Error.EACCES: // match Win32 exception
                         throw new IOException(SR.Format(SR.UnauthorizedAccess_IODenied_Path, sourceFullPath), errorInfo.RawErrno);
                     default:
-                        throw Interop.GetExceptionForIoErrno(errorInfo, sourceFullPath, isDirectory: true);
+                        throw Interop.GetExceptionForIoErrno(errorInfo, isDirectory: true);
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Unix.cs:408]: @tmds, you ok with this?  Seems reasonable, if for no other reason than there are two paths involved and we're sort of arbitrarily picking one to include in the error.

File: src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Windows.cs (hunk 1 above)
Comment (steveberdy) [reply src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Windows.cs]: Done.

Comment (steveberdy) [reply src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Windows.cs]: @stephentoub I've reverted those changes.

File: src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Unix.cs (hunk 2 above)
Comment (tmds) [reply src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Unix.cs:408]: Sure. It makes sense to include both or neither.

Comment (steveberdy) [reply src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Unix.cs:408]: I made it to where there were zero paths included to match what is on the Windows side. Is this able to merge now?

**Review by stephentoub — APPROVED** [review]

**Comment by stephentoub** [comment]
Thanks.

**Pull request merged**